src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/updatecontactdetails/UpdateContactDetailsConsentedMidHandler.java

```diff
@@ -35,12 +57,71 @@ public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType
     @Override
     public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(FinremCallbackRequest callbackRequest,
                                                                               String userAuthorisation) {
-        FinremCaseDetails caseDetails = callbackRequest.getCaseDetails();
+        FinremCaseDetails finremCaseDetails = callbackRequest.getCaseDetails();
         log.info("Invoking consented event {} mid event callback for case id {}", EventType.UPDATE_CONTACT_DETAILS,
-            caseDetails.getId());
+            finremCaseDetails.getId());
+
+        FinremCaseData caseData = finremCaseDetails.getData();
+        List<String> errors = new ArrayList<>();
+        errors.addAll(postalService.validate(caseData));
+        errors.addAll(validateCaseData(caseData));
+
+        Optional<ContactDetailsWrapper> contactDetailsWrapper = Optional.ofNullable(caseData.getContactDetailsWrapper());
```

Comment:
Take a look at the method FinremCaseData.getContactDetailsWrapper(). The return value can never be null.

Reply:
If there are validation errors then should we be continuing on with doing a notice of change?

Reply:
This was copied from the AboutToSubmit handler in error. Have removed it this.